internal/runbits/buildscript/file.go

```diff
@@ -57,42 +55,48 @@ func Initialize(path string, auth *authentication.Auth) error {
 	if err != nil {
 		return errs.Wrap(err, "Unable to get the local commit ID")
 	}
+
 	buildplanner := buildplanner.NewBuildPlannerModel(auth)
-	expr, atTime, err := buildplanner.GetBuildExpressionAndTime(commitId.String())
+	script, err = buildplanner.GetBuildScript(commitId.String())
 	if err != nil {
 		return errs.Wrap(err, "Unable to get the remote build expression and time")
 	}
-	script, err = NewFromBuildExpression(atTime, expr)
+
+	sb, err := script.Marshal()
 	if err != nil {
-		return errs.Wrap(err, "Unable to convert build expression to build script")
+		return errs.Wrap(err, "Unable to marshal build script")
 	}
 
 	logging.Debug("Initializing build script at %s", scriptPath)
-	err = fileutils.WriteFile(scriptPath, []byte(script.String()))
+	err = fileutils.WriteFile(scriptPath, sb)
 	if err != nil {
 		return errs.Wrap(err, "Unable to write build script")
 	}
 
 	return nil
 }
 
-func Update(proj projecter, atTime *strfmt.DateTime, newExpr *buildexpression.BuildExpression) error {
+func Update(proj projecter, newScript *buildscript.BuildScript) error {
 	script, err := ScriptFromProject(proj)
 	if err != nil {
 		return errs.Wrap(err, "Could not read build script")
 	}
 
-	newScript, err := NewFromBuildExpression(atTime, newExpr)
+	equals, err := script.Equals(newScript)
```

Comment:
I kinda like having the logic of this be owned by the buildscript package, even if it is straight forward. Plus having a single Equals() function means having a single error handler. Otherwise we'd need to handle errors for both marshals.